Apps/W1/EDocumentConnectors/Tietoevry/app/src/Events.Codeunit.al

```diff
@@ -0,0 +1,115 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.EServices.EDocumentConnector.Tietoevry;
+
+using Microsoft.Sales.Peppol;
+using Microsoft.Sales.Document;
+using Microsoft.Foundation.Company;
+using Microsoft.Sales.Customer;
+using Microsoft.eServices.EDocument;
+using Microsoft.eServices.EDocument.Service.Participant;
+codeunit 6395 Events
+{
+    SingleInstance = true;
+    EventSubscriberInstance = StaticAutomatic;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"PEPPOL Validation", OnCheckSalesDocumentOnBeforeCheckCompanyVATRegNo, '', false, false)]
+    local procedure "PEPPOL Validation_OnCheckSalesDocumentOnBeforeCheckCompanyVATRegNo"(SalesHeader: Record "Sales Header"; CompanyInformation: Record "Company Information"; var IsHandled: Boolean)
+    var
+        ExternalConnectionSetup: Record "Connection Setup";
+    begin
+        if not ExternalConnectionSetup.Get() then
+            exit;
+
+        IsHandled := true;
+        if CompanyInformation."VAT Registration No." = '' then
+            Error(this.MissingCompInfVATRegNoErr, CompanyInformation.TableCaption());
+    end;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"PEPPOL Validation", OnCheckSalesDocumentOnBeforeCheckCustomerVATRegNo, '', false, false)]
+    local procedure "PEPPOL Validation_OnCheckSalesDocumentOnBeforeCheckCustomerVATRegNo"(SalesHeader: Record "Sales Header"; Customer: Record Customer; var IsHandled: Boolean)
+    var
+        ServiceParticipant: Record "Service Participant";
+        EDocumentService: Record "E-Document Service";
+        EDocExtConnectionSetup: Record "Connection Setup";
+    begin
+        if not EDocExtConnectionSetup.Get() then
+            exit;
+        EDocumentService.SetRange("Service Integration V2", EDocumentService."Service Integration V2"::Tietoevry);
+        if not EDocumentService.FindFirst() then
+            exit;
```

Comment:
Not going to work. There can be multiple services with this set.

Reply:
resolved